File: v6_composite_reuse/java/CarMaker2.java
package v6_composite_reuse.java;

public class CarMaker2 {

    public static void main(String[] args) {
        Car c = new Car(new Gasoline(), new Blue());
        c.move();
    }
    
}

class Car {
    private Energy energy;
    private Color color;
    public Car(Energy e, Color c) {
        this.energy = e;
        this.color = c;
    }
    public void move() {
        System.out.println(energy.toString() + color.toString() + "move");
    }
}

interface Energy {}

class Gasoline implements Energy { 
    public String toString() {return "gasoline";}
}

class Electric implements Energy { 
    public String toString() {return "electric";}
}

interface Color {}

class Red implements Color {
    public String toString() {return "red";}
}

class Blue implements Color {
    public String toString() {return "blue";}
}

File: v6_composite_reuse/python/car_maker2.py
class Car:
    enery = None
    color = None
    def __init__(self, energy, color):
        self.enery = energy
        self.color = color
    
    def move(self):
        print(f'{self.enery.__str__()} {self.color.__str__()}')

class Energy:
    pass

class Color:
    pass

class Gasoline(Energy):
    def __str__(self): return 'gasoline'

class Electric(Energy):
    def __str__(self): return 'electric'

class Red(Color):
    def __str__(self): return 'red'

class Blue(Color):
    def __str__(self): return 'blue'

Car(Gasoline(), Red()).move()
Car(Electric(), Blue()).move()
